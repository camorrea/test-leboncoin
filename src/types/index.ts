export enum DataStatus {
  notAsked = 'NotAsked',
  loading = 'Loading',
  failure = 'Error',
  success = 'Success'
}

export enum Confidentiality {
  public = 'Public',
  private = 'Private'
}

export enum User {
  me = 'Me',
  anonymous = 'Anonymous'
}

export type MessageType = {
  id: string
  text: string
  confidentiality: Confidentiality
  user: User
}
