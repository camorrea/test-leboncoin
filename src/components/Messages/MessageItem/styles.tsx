import React from 'react'
import styled from 'styled-components'

import { Confidentiality, User } from '../../../types'

export const Container = styled.div<{ user: User }>`
  display: flex;
  align-items: flex-start;
  justify-content: ${({ user }) =>
    user === User.me ? 'flex-end' : 'flex-start'};
`

export const Content = styled.div<{ confidentiality: Confidentiality }>`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  padding: 0.5rem;
  margin-bottom: 0.5rem;
  min-height: 1.2rem;
  width: 85%;
  border-radius: 0.6rem;
  background-color: ${({ confidentiality }) =>
    confidentiality === Confidentiality.public ? '#FFFFFF' : '#444444'};
  color: ${({ confidentiality }) =>
    confidentiality === Confidentiality.public ? '#444444' : '#FFFFFF'};
  border: 1px solid
    ${({ confidentiality }) =>
      confidentiality === Confidentiality.public ? '#444444' : '#FFFFFF'};
`

export const Text = styled.p`
  font-size: 0.7rem;
  line-height: 1rem;
`

export const IconContainer = styled.div`
  padding-left: 0.5rem;
  width: 28px;
`

export const PrivateIcon = () => (
  <svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1">
    <g fill="#ffffff" stroke="none">
      <path d="M 7.6875 0.6875 C 5.09375 1.328125 2.59375 3.359375 1.4375 5.78125 C 0.6875 7.359375 0.46875 8.328125 0.46875 10 C 0.46875 11.671875 0.6875 12.640625 1.4375 14.21875 C 2.328125 16.078125 3.921875 17.671875 5.78125 18.5625 C 7.359375 19.3125 8.328125 19.53125 10 19.53125 C 11.671875 19.53125 12.640625 19.3125 14.21875 18.5625 C 16.078125 17.671875 17.671875 16.078125 18.5625 14.21875 C 19.3125 12.640625 19.53125 11.671875 19.53125 10 C 19.53125 8.328125 19.3125 7.359375 18.5625 5.78125 C 17.671875 3.921875 16.078125 2.328125 14.21875 1.4375 C 13.484375 1.09375 12.515625 0.734375 12.0625 0.640625 C 10.96875 0.421875 8.734375 0.4375 7.6875 0.6875 Z M 11.90625 1.421875 C 13.421875 1.734375 14.953125 2.59375 16.171875 3.828125 C 19.65625 7.296875 19.65625 12.703125 16.171875 16.171875 C 14.4375 17.921875 12.453125 18.75 10 18.75 C 7.546875 18.75 5.5625 17.921875 3.828125 16.171875 C 2.078125 14.4375 1.25 12.453125 1.25 10 C 1.25 8.53125 1.484375 7.484375 2.140625 6.109375 C 3.171875 3.9375 5.515625 2.046875 7.859375 1.46875 C 8.890625 1.21875 10.828125 1.203125 11.90625 1.421875 Z M 11.90625 1.421875 " />
      <path d="M 5.96875 8.015625 C 5.78125 8.21875 5.625 8.578125 5.625 8.8125 C 5.625 9.34375 6.265625 10 6.796875 10 C 7.34375 10 7.96875 9.34375 7.96875 8.78125 C 7.96875 7.796875 6.65625 7.296875 5.96875 8.015625 Z M 5.96875 8.015625 " />
      <path d="M 12.375 8.015625 C 12.1875 8.21875 12.03125 8.578125 12.03125 8.8125 C 12.03125 9.34375 12.671875 10 13.203125 10 C 13.75 10 14.375 9.34375 14.375 8.78125 C 14.375 7.796875 13.0625 7.296875 12.375 8.015625 Z M 12.375 8.015625 " />
      <path d="M 11.71875 11.40625 L 9.90625 12.3125 L 7.890625 11.609375 C 6.546875 11.15625 5.78125 10.96875 5.640625 11.0625 C 5.515625 11.125 5.25 11.65625 5.0625 12.234375 C 4.75 13.171875 4.734375 13.3125 4.9375 13.515625 C 5.078125 13.640625 5.234375 13.75 5.3125 13.75 C 5.390625 13.75 5.640625 13.828125 5.890625 13.921875 L 6.328125 14.09375 L 5.59375 14.46875 C 5.171875 14.671875 4.84375 14.921875 4.84375 15 C 4.84375 15.34375 5.8125 17.046875 6.046875 17.140625 C 6.1875 17.203125 7.078125 16.84375 8.21875 16.28125 L 10.125 15.328125 L 12.171875 16.015625 C 13.75 16.546875 14.25 16.65625 14.421875 16.515625 C 14.546875 16.40625 14.796875 15.890625 14.96875 15.359375 C 15.34375 14.234375 15.265625 14.046875 14.296875 13.75 L 13.6875 13.546875 L 14.328125 13.234375 C 14.6875 13.0625 15.03125 12.8125 15.0625 12.703125 C 15.109375 12.59375 14.921875 12.03125 14.65625 11.484375 C 14.28125 10.71875 14.078125 10.46875 13.828125 10.484375 C 13.65625 10.484375 12.703125 10.890625 11.71875 11.40625 Z M 13.984375 11.875 L 14.21875 12.34375 L 10.296875 14.296875 C 8.140625 15.390625 6.359375 16.25 6.328125 16.21875 C 6.3125 16.1875 6.1875 15.953125 6.046875 15.703125 L 5.796875 15.234375 L 9.578125 13.328125 C 11.65625 12.28125 13.453125 11.421875 13.546875 11.421875 C 13.65625 11.40625 13.84375 11.609375 13.984375 11.875 Z M 7.453125 12.296875 C 8.171875 12.53125 8.75 12.765625 8.75 12.8125 C 8.75 12.859375 8.453125 13.046875 8.078125 13.234375 C 7.453125 13.546875 7.390625 13.546875 6.578125 13.296875 C 6.125 13.15625 5.75 13 5.75 12.921875 C 5.78125 12.5625 5.96875 11.875 6.0625 11.875 C 6.109375 11.875 6.734375 12.0625 7.453125 12.296875 Z M 13.859375 14.453125 C 14.34375 14.65625 14.359375 14.6875 14.1875 15.1875 C 14.09375 15.46875 14 15.71875 13.984375 15.734375 C 13.96875 15.78125 12.09375 15.171875 11.40625 14.90625 C 11.25 14.84375 11.390625 14.703125 11.875 14.4375 C 12.609375 14.0625 12.90625 14.0625 13.859375 14.453125 Z M 13.859375 14.453125 " />
    </g>
  </svg>
)
