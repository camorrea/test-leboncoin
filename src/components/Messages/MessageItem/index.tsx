import React from 'react'

import { Confidentiality, MessageType } from '../../../types'

import { Container, IconContainer, PrivateIcon, Text } from './styles'

type Props = {
  message: MessageType
}

const MessageItem = (props: Props) => {
  const { message } = props
  return (
    <Container confidentiality={message.confidentiality}>
      <Text>{message.text}</Text>
      {message.confidentiality === Confidentiality.private && (
        <IconContainer title={message.confidentiality}>
          <PrivateIcon />
        </IconContainer>
      )}
    </Container>
  )
}

export default React.memo(MessageItem)
