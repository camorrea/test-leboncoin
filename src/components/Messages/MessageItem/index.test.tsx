import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import renderer from 'react-test-renderer'

import { Confidentiality, User } from '../../../types'
import MessageItem from './'

configure({ adapter: new Adapter() })

const message = {
  id: 'e5bba129-7b23-61e8-fa2f-00001cadce3f',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  confidentiality: Confidentiality.public,
  user: User.anonymous
}

const component = <MessageItem message={message} />

describe('MessageItem Component', () => {
  it('should render properly', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
