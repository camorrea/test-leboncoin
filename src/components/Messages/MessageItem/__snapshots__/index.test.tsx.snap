// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem Component should render properly 1`] = `
<div
  className="sc-bdfBwQ blNLYb"
>
  <div
    className="sc-gsTCUz feHatn"
  >
    <p
      className="sc-dlfnbm ePHJPa"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit
    </p>
  </div>
</div>
`;
